File: agri-ecommerce-backend/src/main/java/com/agri/ecommerce/service/WishlistService.java
package com.agri.ecommerce.service;

import com.agri.ecommerce.dto.request.wishlist.AddWishlistItemRequest;
import com.agri.ecommerce.dto.response.wishlist.WishlistResponse;

public interface WishlistService {

    WishlistResponse getWishlist(Long userId);

    WishlistResponse addItem(Long userId, AddWishlistItemRequest request);

    WishlistResponse removeItem(Long userId, Long productId);

    WishlistResponse clearWishlist(Long userId);
}

File: agri-ecommerce-frontend/src/services/wishlist.service.js
import axiosClient from "@/lib/axios-client";

const unwrapApiData = (response) => response?.data ?? response;

export const wishlistService = {
  getWishlist: async () => {
    const response = await axiosClient.get("/customer/wishlist");
    return unwrapApiData(response);
  },

  addItem: async (payload) => {
    const response = await axiosClient.post("/customer/wishlist/items", payload);
    return unwrapApiData(response);
  },

  removeItem: async (productId) => {
    const response = await axiosClient.delete(
      `/customer/wishlist/items/${productId}`
    );
    return unwrapApiData(response);
  },
};
